roles/gpu/tasks/gpu.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Install yum requirements for gpu driver installation
+  ansible.builtin.yum:
+    state: 'installed'
+    update_cache: true
+    name:
+      - tar
+      - bzip2
+      - make
+      - automake
+      - gcc
+      - gcc-c++
+      - pciutils
+      - elfutils-libelf-devel
+      - libglvnd-devel
+      - bind-utils
+      - wget
+  become: true
+
+# ansible_kernel variable is not working, as after reboot, still holds old kernel
+- name: Get current kernel version
+  ansible.builtin.command: '/usr/bin/uname -r'
+  register: uname_output
+  failed_when: uname_output.rc != 0
+  when: true
+  become: true
+
+- name: Set kernel version fact
+  ansible.builtin.set_fact:
+    kernel_version: "{{ uname_output.stdout }}"
+
+- name: Install kernel developement package matching running kernel version
+  ansible.builtin.yum:
+    name: 'kernel-devel-{{ kernel_version }}'
+  register: yum_result
+  failed_when: yum_result.rc != 0
+  when: true
+  become: true
+
+- name: Download a driver installation file from NVidia
+  ansible.builtin.get_url:
+    url: '{{ gpu_url_directory }}/{{ gpu_runfile }}'
+    dest: '/root/{{ gpu_runfile }}'
+    mode: '0700'
+  become: true
+
+- name: "Check if driver downloaded"
+  ansible.builtin.stat:
+    path: '/root/{{ gpu_runfile }}'
+  when: true
+  register: driver_downloaded
+  become: true
+
+- name: Install driver from .run file
+  ansible.builtin.command: '/root/{{ gpu_runfile }} --silent --driver'
+  register: install_result
+  failed_when: install_result.rc != 0
+  when: driver_downloaded.stat.exists
+  notify: reboot_server
+  become: true
+
+- name: Remove installation file
```

Comment:
Why remove the installer? That means it will have to be re-downloaded again each time the role is executed.

Reply:
Well, alternative is to use 3.3GB on every GPU machine, even after the installation was successful. If the installation would be small, then it can be easily downloaded- if large, then it needs to be downloaded.
When it comes to this step, the driver should be already installed, as all went fine. And so when this role is finalized, there should be no more reason to keep the file. It either works, or it doesn't and on-machine debugging needs to be done - manually.

Reply:
What I do realize, is that I should have at the end the same (as the one at the beginning) checkup with `nvidia-smi` and check if all devices are listed there.

Reply:
Ok, 3.3 GB is quite a large file for the root partition. In that case I understand why it is removed again. Would be nice though to be able to _cache_ this somehow for a cluster just like we do with RPMs using Pulp. Would be bad if a compute node needs to be re-installed and the version of a driver is no longer available for download for whatever reason.

Reply:
As this GPU related stuff is work in progress anyway, I 'm happy to accept this as is with deleting the driver download, but it would be nice to see if we can use the `file` plugin caching of these driver files with Pulp in a future PR. We already install the `file` plugin in addition to the `rpm` plugin, but have not used it yet.